src/Shared/RoslynUtils/SymbolExtensions.cs

```diff
@@ -82,4 +82,25 @@ public static ImmutableArray<IParameterSymbol> GetParameters(this ISymbol? symbo
 
     public static ISymbol? GetAnySymbol(this SymbolInfo info)
         => info.Symbol ?? info.CandidateSymbols.FirstOrDefault();
+
+    public static bool IsOptional(this IParameterSymbol parameterSymbol) =>
+        parameterSymbol.Type is INamedTypeSymbol
+        {
+            NullableAnnotation: NullableAnnotation.Annotated
+        } || parameterSymbol.HasExplicitDefaultValue;
+
+    public static bool TryGetNamedArgumentValue<T>(this AttributeData attribute, string argumentName, [NotNullWhen(true)] out T? argumentValue)
+    {
+        argumentValue = default;
+        foreach (var namedArgument in attribute.NamedArguments)
+        {
+            if (namedArgument.Key == argumentName)
+            {
+                var routeParameterNameConstant = namedArgument.Value;
+                argumentValue = (T)routeParameterNameConstant.Value!;
```

Comment:
What if the argument is set to `null`? Are we sure this will always be non null?

Reply:
Ah, good call! Yes, it's possible to do something like `[FromQuery(Name = null)]` for example. I updated the implementation here and added a guard against this scenario.